### PR TITLE
fix(eslint-plugin): [no-type-alias] handle Template Literal Types

### LINKED ISSUE #5043: Bug: [no-type-alias] template literals are unhandled
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-this-alias/

### Description

Currently when aliasing [template literal type](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html), an error is thrown with message "Type unhandled are not allowed"

image

This rule should recognise template literal type. Additionally, a new option `allowTemplateLiteral` should be introduced to control rule's behaviour similarly to other options in this rule.

### Fail

```typescript
// with allowTemplateLiteral: false (default)
type SlashCommand = `/${string}`
```

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #5043

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/no-type-alias.md [hunk 1]
```diff
@@ -118,6 +118,9 @@ type Foo = string | string[];
 
 type Foo = string & string[];
 
+// Template Literal Types
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/docs/rules/no-type-alias.md:121]: Nit: we generally don't explain type names like this. The rest of the rule's docs don't.

```suggestion
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great to me, thanks @a-tarasyuk! 

Just the one nitpick on removing all the `// Template Literal Types` comments. I can that change if you don't mind them being removed, or just wait for you to. Whichever's easier! 🙂

File: packages/eslint-plugin/docs/rules/no-type-alias.md (hunk 1 above)
Comment (a-tarasyuk) [reply packages/eslint-plugin/docs/rules/no-type-alias.md]: @JoshuaKGoldberg Sure. I've removed comments.

**Review by JoshuaKGoldberg — APPROVED** [review]
Yup, looks great - thanks!

**Pull request merged**